packages/plugin-ext/src/common/plugin-protocol.ts

```diff
@@ -585,6 +585,25 @@ export interface MetadataProcessor {
     process(pluginMetadata: PluginMetadata): void;
 }
 
+/**
+ * Designed to retreive plugins resources from non default or remote location.
+ */
+export const PluginResourcesProvider = Symbol.for('ResourcesProvider');
+export interface PluginResourcesProvider {
```

Comment:
`Resource` and `ResourcProvider` already have a meaning in Theia, could you come with other names?

Reply:
Maybe PluginResourceContentProvider?

Reply:
@akosyakov I think yes, but to me it is better to have name as close to the meaning as possible. I gave this name here because _plugin resource_ is general word for images, scripts, styles, snippets etc. which a plugin has.
In addition to @AndrienkoAleksandr's proposal, I may suggest `PluginDataProvider`. Don't know which one is better. 
WDYT?

Reply:
Could it be moved to the Che extension since the vanilla Theia does not seem to support remote extensions? It is hard to judge for me how it should be named and work since i cannot run if from examples.

I would be fine with doing something like  https://github.com/theia-ide/theia/pull/6070/files#r319754377, i.e.
```ts
app.get('/hostedPlugin/:pluginId/:path(*)', async (req, res) => {
    // ...
    if (localPath) {
        // ...
    } else {
        this.handleMissingFile({pluginId, filePath, req, res});
    }
});

protected handleMissingResource(...): void {
    res.status(404).send(`The plugin with id '${escape_html(pluginId)}' does not exist.`);
}
```

in the che extension you subclass `HostedPluginReader` and override `handleMissingFile``, without any unused new APIs in the vanilla Theia

Alternatively it would be good to setup the browser example with remote extensions that any committer can test and reason about it from sources. If it is not feasible then it is not in the scope of the project. I generally think remote extensions is valid case, but in the current setup is hard to work on them. I've added a point to an agenda for tomorrow.